test/unit_tests/js/modules/util/validators-spec.js

```diff
@@ -16,26 +21,29 @@ describe( 'Validators', () => {
       expect( returnedObject ).toStrictEqual( {} );
     } );
 
-    it( 'should allow single digit for month', () => {
-      testField.value = '1/22/2017';
+    it( 'should return an error object for a malformed date', () => {
+      testField.value = '11.12.2007';
       returnedObject = validators.date( testField );
 
-      expect( returnedObject ).toStrictEqual( {} );
+      expect( returnedObject['date'] ).toBe( false );
+      expect( returnedObject['msg'] ).toBe( ERROR_MESSAGES.DATE.INVALID );
     } );
 
-    it( 'should allow single digit for day', () => {
-      testField.value = '11/2/2017';
-      returnedObject = validators.date( testField );
-
-      expect( returnedObject ).toStrictEqual( {} );
+    it( 'should allow correctly formatted dates', () => {
+      for (let i = 0; i < validDates.length; i++) {
+        testField.value = validDates[i];
+        returnedObject = validators.date( testField );
+        expect( returnedObject ).toStrictEqual( {} );
+      }
     } );
 
-    it( 'should return an error object for a malformed date', () => {
-      testField.value = '11-12-2007';
-      returnedObject = validators.date( testField );
-
-      expect( returnedObject['date'] ).toBe( false );
-      expect( returnedObject['msg'] ).toBe( ERROR_MESSAGES.DATE.INVALID );
+    it( 'should reject incorrectly formatted dates', () => {
+      for (let i = 0; i < invalidDates.length; i++) {
```

Comment:
Is this still how we iterate over arrays in JS? I suspect things have changed since the last time I wrote javascript 4 years ago.

Reply:
That's fine actually. There are newer loops, but an old for loop is fine and will be guaranteed browser compatibility. The one optimization I usually make is:

```js
for (let i = 0, len = invalidDates.length; i < len; i++) {
```
Which should only look up the length once, not once per loop. Though I haven't benchmarked this.